### PR TITLE
Add well-formed check for undefined values

### PR DESCRIPTION
**Pull request by jacob-hughes**
This checks that operand values inside each JIT IR instruction refers to a valid SSA variable.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs [hunk 1]
```diff
@@ -22,12 +22,22 @@
...
 
 use super::{BinOp, BinOpInst, Const, GuardInst, Inst, Module, Operand, Ty};
 
 impl Module {
     pub(crate) fn assert_well_formed(&self) {
         for (iidx, inst) in self.iter_skipping_insts() {
+            inst.map_packed_operand_locals(self, &mut |x| {
+                let i = self.inst_all(x);
+                if i.def_type(self).is_none() {
+                    panic!(
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs:35]: Can we add a `#[should_panic]` test for this?

Comment (jacob-hughes) [reply ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs:35]: I don't think we can. I had a go but it looks like this sort of error is already caught by the parser:

`.../jit_ir/parser.rs:559:  self.error_at_span(span, &format!("Undefined local operand '%{idx}'"))`

So I think the test suite is too clever for me to build this kind of malformed IR

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs:35]: Indeed, I did think I'd implemented this somewher! One could create the mall-formed IR by hand, but it's annoying to do so. I think this is OK as-is.

**Comment by jacob-hughes** [comment]
Not sure I understand this CI failure. Have you seen this before?

```
22:34:57 Fetching submodule ykllvm
22:34:58 fatal: remote error: upload-pack: not our ref 6b5702f38ad9355c7af836eb5007f94dca8611f3
22:34:58 Errors during submodule fetch:
22:34:58 ykllvm
```

**Comment by ltratt** [comment]
Nope. Maybe it's transient. Let's try again.

**Comment by ltratt** [comment]
Might be one for @vext01.

**Comment by vext01** [comment]
I've pushed a commit that may fix this, although it's untested.

It's a mystery where that bogus git hash comes from. It may have to do with shallow clones, but I honestly don't know.

**Comment by ltratt** [comment]
Thanks -- let's try it!

**Comment by vext01** [comment]
(btw, jake didn't sign his commit)

**Comment by ltratt** [comment]
@jacob-hughes Please force push a signed commit.

**Comment by jacob-hughes** [comment]
Force pushed a signed commit

**Pull request merged**